libraries/profile.rb

```diff
@@ -61,7 +62,7 @@ class ComplianceProfile < Chef::Resource # rubocop:disable Metrics/ClassLength
 
       # retrieve access token if a refresh token is set
       access_token = token
-      access_token = retrieve_access_token unless refresh_token.nil?
+      access_token = retrieve_access_token(server, refresh_token) unless refresh_token.nil?
```

Comment:
@chris-rock @srenatus I agree completely.  I've added a `node['audit']['insecure']` attribute and set the default to `false` in the library.